source/images.py

```diff
@@ -0,0 +1,129 @@
+from typing import List
+
+import numpy as np
+from PIL import Image
+
+
+class Images:
+    """Module to handle image conversion and processing"""
+
+    def __init__(self, file: str,
+                 cols: int = 80,
+                 scale: float = 0.43,
+                 resolution: int = 0,
+                 shade_str: str = "$@B%8&WM#*oahkbdpqwmZO0QLCJUYXzcvunxrjft/\\()1{}[]?-_+~<>i!lI;:,\"^`'.",
+                 shade_min: str = "@#+-."):
+        """
+        Initiates the Images class while takes an image file and converts it into an Ascii image
+
+        :param file: The image file path, allows .jpg, .jpeg, .png and other popular file formats
+        Ascii related parameters =>
+        :param cols: Number of columns in the ascii image, defaults to 80
+        :param scale: defines the adjusted aspect ratio for the ascii image, defaults to 0.43
+        :param resolution: defines resolution, takes in values as percentage of the highest resolution possible with
+               characters from shade_str
+               Special value: 0 (default) is interpreted as mode switch to minimalistic output with characters from
+               shade_min
+        :param shade_str: defaults to a well known pattern of 69 characters used to define the greyscale
+        :param shade_min: defaults to a minimalistic pattern of 5 characters used to define the greyscale
+        """
+        # TODO: error handling when no file is selected
+        self.file = file
+        self.image = Image.open(file).convert("L")
+        self.cols = cols
+        self.scale = scale
+        self.shade = {
+            "str": shade_str,
+            "str_len": len(shade_str),
+            "min": shade_min,
+            "min_len": len(shade_min),
+            "length": 0,
+            "resolution": 0
+        }
+
+        # ensure level is [0-100]
+        resolution = 100 if resolution > 100 else 0 if resolution < 0 else resolution
+        self.shade["resolution"] = resolution
+
+    def __repr__(self) -> str:
+        return f'<Images file={self.file} image={self.image}>'
```

Comment:
Unwise to include the whole image like that, if we ever print `__repr__` this will flood the console.
```suggestion
        return f'<Images file={self.file} shade={self.shade}>'
```

Reply:
self.image returns PIL Image instance, I checked it

Reply:
Ah, makes sense.